packages/core/src/dispatcher.ts

```diff
@@ -74,18 +74,24 @@ export default class Dispatcher extends EventEmitter {
    *
    * Logs error if subscribe fails.
    */
-  async _subscribe(): Promise<void> {
+  async _subscribe(force = false): Promise<void> {
     try {
-      await this._ipfs.pubsub.subscribe(
-        this.topic,
-        this.handleMessage,
-        // {timeout: IPFS_GET_TIMEOUT} // ipfs-core bug causes timeout option to throw https://github.com/ipfs/js-ipfs/issues/3472
-      )
-      this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
+      if (force || !(await this._ipfs.pubsub.ls()).includes(this.topic)) {
+        await this._ipfs.pubsub.unsubscribe(this.topic, this.handleMessage)
+        await this._ipfs.pubsub.subscribe(
+          this.topic,
+          this.handleMessage,
+          // {timeout: IPFS_GET_TIMEOUT} // ipfs-core bug causes timeout option to throw https://github.com/ipfs/js-ipfs/issues/3472
+        )
+        this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
+      }
     } catch (error) {
       // TODO: use logger
       if (error.message.includes('Already subscribed')) {
         this.logger.debug(error.message)
+      } else if (error.message.includes('The user aborted a request')) {
```

Comment:
Never seen this. Is it on node 14?

Also should we be using latest IPFS http client?

Reply:
I've used node 14 when testing yes. 

Not updating to latest ipfs-http-client because it might break other stuff (CIDs, multiformats).